### PR TITLE
Rename Discover to Category

### PR DESCRIPTION
**Pull request by eugenes78**
### Description
Rename Discover to Category

### Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have updated the `CHANGELOG` including this PR (where applicable)
- [x] I have run tests and automatic checks locally
- [x] I have run `pitest` check locally and checked that the coverage increased (or didn't change) or decreased insignificantly
- [ ] I have made corresponding changes to the documentation (where applicable)
- [x] I have grouped commits logically or I promise to squash my commits before merge

### CODE REVIEW

File: MapboxSearch/category/api/api-metalava.txt [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    field public static final com.mapbox.search.category.CategoryQuery.Category MOVIE_THEATERS;
+    field public static final com.mapbox.search.category.CategoryQuery.Category MUSEUMS;
+    field public static final com.mapbox.search.category.CategoryQuery.Category NIGHT_CLUBS;
+    field public static final com.mapbox.search.category.CategoryQuery.Category PARKING;
+    field public static final com.mapbox.search.category.CategoryQuery.Category PARKS;
+    field public static final com.mapbox.search.category.CategoryQuery.Category PHARMACY;
+    field public static final com.mapbox.search.category.CategoryQuery.Category RAILWAY_STATION;
+    field public static final com.mapbox.search.category.CategoryQuery.Category RESTAURANTS;
+    field public static final com.mapbox.search.category.CategoryQuery.Category SHOPPING_MALLS;
+    field public static final com.mapbox.search.category.CategoryQuery.Category SUPERMARKET_GROCERY;
```
[the remark this answers is not in the document]
Comment (aokj4ck) [reply MapboxSearch/category/api/api-metalava.txt:90]: @antonvolkau I discussed adding some functionality for Categories and you asked for this to be shared functionality. We can discuss this more offline but I would like Android, iOS and the Search Core SDK to use the same structure for Categories in the Search Core SDK.

I want us to use something like the [existing iOS SearchUI.SearchCategory](https://github.com/mapbox/mapbox-search-ios/blob/main/Sources/MapboxSearchUI/SearchCategory.swift#L6) inside the shared library:

```swift
struct SearchCategory { // todo: rename to Category
    let iconName: String // name
    let legacyName: String // for geocoding
    let canonicalId: String // for SBS, search-box
    let name: String // user-facing display name
}
```

**Review by globaltrouble — APPROVED** [review]

**Comment by eugenes78** [comment]
> What about changelog? Note that this is a breaking change.

Absolutely, it'll be updated as well as the Release Notes.

File: MapboxSearch/category/api/api-metalava.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (eugenes78) [reply MapboxSearch/category/api/api-metalava.txt:90]: @aokj4ck mind creating a ticket for this follow-up work? Unless you think it's a blocker for the current PR.

[the remark this answers is not in the document]
Comment (aokj4ck) [reply MapboxSearch/category/api/api-metalava.txt:90]: Non-blocker!

**Pull request merged**